app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java

```diff
@@ -0,0 +1,87 @@
+package io.neurolab.main.task;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import io.neurolab.main.NFBServer;
+import io.neurolab.model.Task;
+
+public class BinaryFileOutputTask implements Task {
+
+    private static final byte[] FileOutputStream = null;
+    private NFBServer nfbServer;
+    private FileOutputStream out = null;
+    private byte[] stopBytes;
+    private String fileName = "defaultFileOutput.bw";
+    private boolean write = false;
+
+    public BinaryFileOutputTask(NFBServer nfbServer) {
+        this.nfbServer = nfbServer;
+        this.stopBytes = doubleToByteArray(Double.MAX_VALUE);
+    }
+
+    public static byte[] doubleToByteArray(double value) {
+        byte[] bytes = new byte[8];
+        ByteBuffer.wrap(bytes).putDouble(value);
+        return bytes;
+    }
+
+    public static byte[] longToByteArray(long value) {
+        byte[] bytes = new byte[8];
+        ByteBuffer.wrap(bytes).putLong(value);
+        return bytes;
+    }
+
+    public static double toDouble(byte[] bytes) {
+        return ByteBuffer.wrap(bytes).getDouble();
+    }
+
+    public void setFileName(String fileName) {
+        this.fileName = fileName;
+        init();
+    }
+
+    public void setWrite(boolean write) {
+        this.write = write;
+    }
+
+    @Override
+    public void run() {
+        if ((write) && (!nfbServer.getInputData().isEmpty())) {
+            try {
```

Comment:
@suresh-gandhi Motive of changes is good. Use try with resources.

Reply:
out is a global variable used in a couple of more functions init and stop (which are necessary to implement being a part and parcel of the Task Interface) and hence we would be requiring an update to Java 9 to use try with resources here or other option would be to use finally block.

Reply:
Well, I have implemented try with resources using Java 8.

Reply:
@jdeep you can't do it with out-of-scope variables in Java-8. You must have done with in-scope variables.

Reply:
@jddeep I am adding the finally block and updating the PR as I believe this is the best we can do.

Reply:
@suresh-gandhi Alright. Yes you can add the finally block here.